### PR TITLE
M3 work

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: Model3/ModelV3/Context.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿namespace NetDaemon.Common.ModelV3
+{
+    /// <summary>
+    ///     Context
+    /// </summary>
+    public class Context
```
Comment (helto4real) [inline Model3/ModelV3/Context.cs:6]: Why a new class? Is it to decouple from NetDaemon.Common?

Comment (FrankBakkerNl) [reply Model3/ModelV3/Context.cs:6]: We could reuse it here, but indeed I wanted to try to have minimal deoendencies

File: Model3/ModelV3/NDApplication.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+using Microsoft.Extensions.Logging.Abstractions;
+using Model3.ModelV3;
+using NetDaemon.Common.Reactive;
+using NetDaemon.Daemon;
+using NetDaemon.Mapping;
+
+namespace NetDaemon.Common.ModelV3
+{
+    // TODO: try to derive from a simpler base class or interface if possible so we require as little as possible
+    public class NdApplication : NetDaemonAppBase, IHandleHassEvent, IHaContext
```
Comment (helto4real) [inline Model3/ModelV3/NDApplication.cs:16]: So basically you are going for a small subset of the current API? Not store state etc or is it just not implemented?

**Comment by FrankBakkerNl** [comment]
Superseded by #434

**Pull request closed without merging**